WalletWasabi/Blockchain/Keys/WalletGenerator.cs

```diff
@@ -25,28 +25,35 @@ public WalletGenerator(string walletsDir, Network network)
 	public Network Network { get; private set; }
 	public uint TipHeight { get; set; }
 
-	public (KeyManager, Mnemonic) GenerateWallet(string walletName, string password)
+	public (KeyManager, Mnemonic) GenerateWallet(string walletName, string password, Mnemonic? mnemonic = null)
```

Comment:
Diff is ugly here, but all that happened is (1) extraction of a function: 

```cs
string walletFilePath = GetWalletFilePath(walletName, WalletsDir);
```

(2) and using the mnemonic if provided:

```cs
var km = mnemonic is null ? KeyManager.CreateNew(out mnemonic, password, Network) : KeyManager.CreateNew(mnemonic, password, Network);
```